observability/micrometer/src/main/java/org/wildfly/extension/micrometer/MicrometerSubsystemAdd.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ *
+ * Copyright 2021 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wildfly.extension.micrometer;
+
+import static org.jboss.as.controller.OperationContext.Stage.RUNTIME;
+import static org.jboss.as.controller.OperationContext.Stage.VERIFY;
+import static org.jboss.as.controller.PathAddress.EMPTY_ADDRESS;
+import static org.jboss.as.server.deployment.Phase.DEPENDENCIES;
+import static org.jboss.as.server.deployment.Phase.POST_MODULE;
+import static org.jboss.as.server.deployment.Phase.POST_MODULE_METRICS;
+import static org.wildfly.extension.micrometer.MicrometerSubsystemDefinition.METRICS_SCAN_CAPABILITY;
+import static org.wildfly.extension.micrometer.MicrometerSubsystemDefinition.MICROMETER_COLLECTOR;
+import static org.wildfly.extension.micrometer.MicrometerSubsystemExtension.SUBSYSTEM_NAME;
+
+import java.util.List;
+import java.util.function.Function;
+
+import org.jboss.as.controller.AbstractBoottimeAddStepHandler;
+import org.jboss.as.controller.OperationContext;
+import org.jboss.as.controller.OperationFailedException;
+import org.jboss.as.controller.registry.ImmutableManagementResourceRegistration;
+import org.jboss.as.controller.registry.Resource;
+import org.jboss.as.server.AbstractDeploymentChainStep;
+import org.jboss.as.server.DeploymentProcessorTarget;
+import org.jboss.dmr.ModelNode;
+import org.jboss.msc.service.ServiceController;
+import org.wildfly.extension.micrometer.deployment.MicrometerSubsystemDeploymentProcessor;
+import org.wildfly.extension.micrometer.metrics.MicrometerCollector;
+
+public class MicrometerSubsystemAdd extends AbstractBoottimeAddStepHandler {
+    MicrometerSubsystemAdd() {
+        super(MicrometerSubsystemDefinition.ATTRIBUTES);
+    }
+
+    public static final MicrometerSubsystemAdd INSTANCE = new MicrometerSubsystemAdd();
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void performBoottime(OperationContext context, ModelNode operation, ModelNode model)
+            throws OperationFailedException {
+        // If the MP Metrics module is not installed, we need to install the WF Metrics DPU and initiate a metrics
```

Comment:
Minor comment change. Shouldn't it be "DUP" deployment unit processor?

Reply:
It should, yes. I have a follow up JIRA filed (https://issues.redhat.com/browse/WFLY-17050). I'll clean that up as part of that effort.